File: interdispy/ch13/ex_010337_random_walker.py
import sys
import random

n = int(sys.argv[1])
size = n * 2
x = n
y = n
count_steps = 0
while True:
    direction = random.randint(0, 4)
    if direction == 0:
        x -= 1
    elif direction == 1:
        y -= 1
    elif direction == 2:
        x += 1
    else:
        y += 1
    print("(%d, %d)" % (x, y))
    count_steps += 1
    if x < 0 or x >= size or y < 0 or y >= size:
        print('Walker hits the boundary')
        print("Walker made %d steps" % count_steps)
        break

File: interdisjv/src/main/java/org/daydevjv/interdisjv/ch13/Ex010337RandomWalker.java
package org.daydevjv.interdisjv.ch13;

import java.util.Random;

public class Ex010337RandomWalker {
    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        int size = n * 2;
        int x = n;
        int y = n;
        Random random = new Random();
        int countSteps = 0;
        while (true) {
            int direction = random.nextInt(4);
            switch (direction) {
                case 0: x--; break;
                case 1: y--; break;
                case 2: x++; break;
                case 3: y++; break;
            }
            System.out.printf("(%d, %d)%n", x, y);
            countSteps++;
            if (x < 0 || x >= size || y < 0 || y >= size) {
                System.out.println("Walker hits the boundary.");
                System.out.printf("Walker made %d steps.%n", countSteps);
                break;
            }
        }
    }
}
